cli/command/swarm/init.go

```diff
@@ -45,21 +48,84 @@ func newInitCommand(dockerCli command.Cli) *cobra.Command {
 	flags.BoolVar(&opts.forceNewCluster, "force-new-cluster", false, "Force create a new cluster from current state")
 	flags.BoolVar(&opts.autolock, flagAutolock, false, "Enable manager autolocking (requiring an unlock key to start a stopped manager)")
 	flags.StringVar(&opts.availability, flagAvailability, "active", `Availability of the node ("active"|"pause"|"drain")`)
+	flags.StringVar(&opts.defaultAddrPool, flagDefaultAddrPool, "", "List of default subnet addresses followed by subnet size (format: <subnet,subnet,..:subnet-size)")
+	flags.SetAnnotation(flagDefaultAddrPool, "version", []string{"1.39"})
 	addSwarmFlags(flags, &opts.swarmOptions)
 	return cmd
 }
 
+// getDefaultAddrPool extracts info from address pool string and fills PoolsOpt struct
+func getDefaultAddrPool(defaultAddrPool string) ([]string, int, error) {
+	var (
+		size int
+		err  error
+	)
+	if defaultAddrPool == "" {
+		// defaultAddrPool is not defined
+		return nil, 0, nil
+	}
+
+	result := strings.Split(defaultAddrPool, ":")
+	if len(result) > 2 {
+		return nil, 0, fmt.Errorf("Invalid default address pool format. Expected format CIDR[,CIDR]*:SUBNET-SIZE")
+	}
+	// if size is not specified default size is 24
```

Comment:
We should make sure that a default is set daemon-side; at the API level, defaults should be useful, so sending `0` should make the daemon pick the default value.

Reply:
This check will go away with new CLI syntax . But will add extra check and setting default value in moby along with above change you suggested